### PR TITLE
Visualiser updates

### PR DESCRIPTION
**Pull request by rowleya**
Updates to the external device library to bring it in line with the changes to SpyNNaker, and to add support for stepped simulations from outside of the tools e.g. for MUSIC integration.  The stepping integration requires SpiNNakerManchester/SpiNNFrontEndCommon#845.

Testing, including additional integration testing with this module, is here:
SpiNNakerManchester/IntegrationTests#71

### CODE REVIEW

**Comment by rowleya** [comment]
I should add that I also added testing and various code style checks and license checks here, hence why so many files have changed...

File: LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# License Agreement
+
+Up to date information for the whole [SpiNNakerManchester Projects](https://github.com/SpiNNakerManchester) can be found [here](http://spinnakermanchester.github.io/common_pages/4.0.0/LicenseAgreement.html)
```
Comment (andrewgait) [inline LICENSE.md:3]: This and subsequent URLs should probably point to 6.0.0 rather than 4.0.0.

**Review by andrewgait — APPROVED** [review]
Again looks fine, just a minor comment about up to date URLs.

File: LICENSE.md (hunk 1 above)
Comment (rowleya) [reply LICENSE.md]: Ah copy paste error!  Fixed now.

**Pull request merged**